### PR TITLE
Add DriverAvailability Index Page for Admin User

### LINKED ISSUE #15: Driver Availability #6: Add Admin Pages
Prerequisites: #10 #11 #12 #13 #14, part of #4 

### 6th issue: Admin Pages

- [x] There are page components that meet all of the acceptance criteria from the admin perspective (index)
- [x] Storybook entries

### PR DESCRIPTION
**Pull request by gretchenlam**
In this PR, I added an admin driver availability index page as shown below (accessible through Users > Driver Availabilities):

Screenshot 2024-03-12 at 6 05 13 PM

The page showcases all driver availabilities as well as the option to review each one, which takes the user to the Edit Driver Availability page. 

I also made changes to Zeel's DriverAvailabilityTable and tests so as to accommodate for admin users being able to review availabilities.

In addition, I edited James's DriverAvailabilityIndexPage and tests so that the regular index page could display all availabilities, not just the first one.

Storybook link: https://ucsb-cs156-w24.github.io/proj-gauchoride-w24-7pm-1/prs/51/storybook/?path=/story/pages-drivers-driveravailabilityindexpageadmin--three-items-admin-user

Closes #15

### CODE REVIEW

**Review by jamespflaging — DISMISSED** [review]
LGTM - tested locally

**Review by Daoyijiucsb — DISMISSED** [review]
LGTM! it also works on dokku

File: frontend/src/main/pages/Drivers/DriverAvailabilityIndexPageAdmin.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    //             <Button
+    //                 variant="primary"
+    //                 href="/driverAvailability/create"
+    //                 style={{ float: "right" }}
+    //             >
+    //                 Create Driver Availability
+    //             </Button>
+    //         )
+    //     } 
+    // }
```
Comment (andrewpengucsb) [inline frontend/src/main/pages/Drivers/DriverAvailabilityIndexPageAdmin.js:34]: Remove commented code, other than that its good

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]

File: frontend/src/main/pages/Drivers/DriverAvailabilityIndexPageAdmin.js (hunk 1 above)
Comment (gretchenlam) [reply frontend/src/main/pages/Drivers/DriverAvailabilityIndexPageAdmin.js]: resolved!

**Review by Ze-el — APPROVED** [review]
LGTM! Removed comments from Driver Availability Index Page

**Review by andrewpengucsb — APPROVED** [review]

**Pull request merged**